test/integration/configuration/test_profile_plugin.py

```diff
@@ -27,6 +29,25 @@ def test_remove_plugin_file(self):
         c.run("profile show", assert_error=True)
         assert "ERROR: The 'profile.py' plugin file doesn't exist" in c.out
 
+    def test_remove_tools_host(self):
+        tc = TestClient(light=True)
+        profile_plugin = textwrap.dedent("""\
+            import subprocess
+            def profile_plugin(profile, **kwargs):
+                if kwargs.get("context") == "build":
+                    if subprocess.getstatusoutput("cmakee")[0] != 0:
+                        profile.tool_requires["!cmake/*"] = ["cmake/[*]"]
```

Comment:
I am not sure I like this, or is good practice.
Using the profile plugin to inject or remove tool-requires seems like not a great platform hygiene. At some point users need to decide what is provided by Conan and what not, but this kind of digs deeper in the "auto-detection" that we have been trying to avoid as much as possible in Conan 2.